utils/test_helper.py

```diff
@@ -0,0 +1,293 @@
+    return test_judge_client
+
+def get_test_normal_user_client(index=0):
+    test_user_client = Client()
+    text = "test_user%s" % (index)
+    test_user_client.login(username=text, password=text)
+    return test_user_client
+
+def create_problem(owner, pname=None, visible=False):
+    if pname == None:
+        pname = random_word(20)
+    problem = Problem.objects.create(pname=pname, owner=owner, visible=visible)
+    return problem
+
+def random_word(length):
+    return ''.join(random.choice(string.lowercase) for i in range(length)) 
+
+def create_judge_code(prefix, pid, file_ex, size=100):
+    file_name = "%s%s%s%s" % (TEST_PATH, prefix, pid, file_ex)
+    try:
+        with open(file_name, 'w') as fp:
+            fp.write(random_word(size))
+    except (IOError, OSError):
+        print "Failed to create judge code for testing..."
+    return file_name
+
+def create_testcase_files(file_name, size=100, uploaded=False):
+    if file_name == "":
+        file_name = random_word(50)
+    if not uploaded:
+        path = TEST_PATH
+    else:
+        path = TESTCASE_PATH
+    input_file_name = "%s%s.in" % (path, file_name)
+    output_file_name = "%s%s.out" % (path, file_name)
+    try:
+        with open(input_file_name, 'w') as t_in:
+            t_in.write(random_word(size))
+        with open(output_file_name, 'w') as t_out:
+            t_out.write(random_word(size))
+    except (IOError, OSError):
+        print "Failed to create testcase files for testing..."
+    return file_name
+
+def create_testcase(problem, time_limit=1, memory_limit=32, local_files=True, uploaded_files=False):
+    testcase = Testcase.objects.create(problem=problem, time_limit=time_limit,
+                                       memory_limit=memory_limit)
+    if local_files:
+        create_testcase_files(testcase.pk)
+    if uploaded_files:
+        create_testcase_files(testcase.pk, uploaded=True)
+    return testcase
+
+def remove_file_if_exists(file_name):
+    try:
+        if os.path.isfile(file_name):
+            os.remove(file_name)
+    except (IOError, OSError):
+        print "Something went wrong when removing file: %s" % (file_name)
+
+def remove_testcase_file(local_file_name, uploaded_file_name=None):
+    local_input_file_name = "%s%s.in" % (TEST_PATH, local_file_name)
+    local_output_file_name = "%s%s.out" % (TEST_PATH, local_file_name)
+    remove_file_if_exists(local_input_file_name)
+    remove_file_if_exists(local_output_file_name)
+    if uploaded_file_name != None:
+        uploaded_input_file_name = "%s%s.in" % (TESTCASE_PATH, uploaded_file_name)
+        uploaded_output_file_name = "%s%s.out" % (TESTCASE_PATH, uploaded_file_name)
+        remove_file_if_exists(uploaded_input_file_name)
+        remove_file_if_exists(uploaded_output_file_name)
+
+def compare_local_and_uploaded_file(local_file_name, uploaded_file_name):
+    try:
+        return filecmp.cmp(local_file_name, uploaded_file_name)
+    except (IOError, OSError):
```

Comment:
Django runs these tests and report how many **failures** and **errors**.
At that time, I thought in this case it could be classified into **failures** because incorrect implementation of NTHUOJ may lead to uploaded file not found.
But now, I think raising an error is better. Thanks.